Add convenience protocol for enums

tholo opened the pull request:
Allow enums to be easily saved and restored from native types,
modeled on implementation from SQLite.

tholo committed:
Add convenience protocol for enums

Allow enums to be easily saved and restored from native types,
modeled on implementation from SQLite.

calebkleveter commented on Sources/FluentMySQL/MySQLType.swift:
```diff
@@ -29,3 +29,55 @@ extension MySQLText: MySQLColumnDefinitionStaticRepresentable {
+
+    /// See `MySQLDataConvertible.convertFromMySQLData(_:)`
+    public static func convertFromMySQLData(_ data: MySQLData) throws -> Self {
+        guard let e = try self.init(rawValue: .convertFromMySQLData(data)) else {
```
This constant should have a different name. Maybe `instance` or `extractedCase`? Just something that is more descriptive.

tholo replied:
Seems reasonable. Someone should likely go back and do that in the SQLite enum code where this was stol... err... got inspiration from?

calebkleveter commented on Sources/FluentMySQL/MySQLType.swift:
```diff
@@ -29,3 +29,55 @@ extension MySQLText: MySQLColumnDefinitionStaticRepresentable {
+/// Provides a default `MySQLDataConvertible` implementation where the type is also
+/// `RawRepresentable` by a `MySQLDataConvertible` type.
+extension MySQLDataConvertible
+    where Self: RawRepresentable, Self.RawValue: MySQLDataConvertible
```
The line break here should probably be removed. It doesn't match the style of the other declarations.

tholo replied:
Yeah -- I think that was a left-over due to the original (SQLite) variant had some much longer things and "had" to be on two lines. But that is no longer the case here...

tholo committed:
Remove extraneous line break

This is not long enough to require a line break, unlike the
original it was taken from.

tholo committed:
Use better variable name

Use a descriptive name for an assigned constant instead of a single
letter name.

tanner0101 reviewed (approved):
Looks great, thanks!

penny-coin commented:
Hey @tholo, you just merged a pull request, have a coin! 

You now have 1 coins.
